src/blueprints/tv2/helpers/tv2-show-style-blueprint-configuration-mapper.ts

```diff
@@ -81,14 +87,32 @@ export class Tv2ShowStyleBlueprintConfigurationMapper {
       graphicsSchemas: this.mapGraphicsSchemas(coreConfiguration.GfxSchemaTemplates),
       selectedGraphicsSetup: this.findSelectedGraphicsSetup(coreConfiguration.GfxDefaults, coreConfiguration.GfxSetups),
       splitScreenConfigurations: this.mapSplitScreenConfigurations(coreConfiguration.DVEStyles),
-      breakerTransitionEffectConfigurations: this.mapTransitionEffectConfigurations([
-        ...coreConfiguration.Transitions.map(transition => transition.Transition),
-        coreConfiguration.ShowstyleTransition
-      ]),
+      // SOF-2004 - Need to handle undefined values from Sofie core
+      // breakerTransitionEffectConfigurations: this.mapTransitionEffectConfigurations([
+      //   ...coreConfiguration.Transitions.map(transition => transition.Transition),
+      //   coreConfiguration.ShowstyleTransition
+      // ]),
+      breakerTransitionEffectConfigurations: this.mapBreakerTransitionEffectConfigurations(
+        coreConfiguration.Transitions, 
+        coreConfiguration.ShowstyleTransition),
       breakers: this.mapToBreakers(coreConfiguration.BreakerConfig)
     }
   }
 
+  private mapBreakerTransitionEffectConfigurations(coreTransitions: Transition[], showstyleTransition: string) : BreakerTransitionEffect[] {
+    //SOF-2004 insert guard clause to protect against undefined values
+    if (!coreTransitions) {
+      return []
+    }
+
+    const transitions = coreTransitions.map(transition => transition.Transition).filter(transition => transition !== undefined)
+
+    // If no undefined value is found, proceed with mapping
+    const breakerTransitionEffectConfigurations = this.mapTransitionEffectConfigurations([...transitions, showstyleTransition])
+    
+    return breakerTransitionEffectConfigurations
```

Comment:
Same comment as for `transitions`. It's not clear what the type of `breakerTransitionEffectConfiugrations` is, so a type should be added.

In this scenario you could just return the result of `this.mapTransitionEffectConfigurations(...)` directly so you don't have to create a local variable that "isn't used".